python-sdk/src/astro/sql/operators/base_decorator.py

```diff
@@ -193,42 +193,52 @@ def translate_jinja_to_sqlalchemy_template(self, context: dict) -> None:
         if context:
             self.sql = self.render_template(self.sql, context)
 
-    def get_openlineage_facets(self, task_instance):
+    def get_openlineage_facets_on_complete(self, task_instance):
         """
         Returns the lineage data
         """
         from astro.lineage import (
             BaseFacet,
             DataSourceDatasetFacet,
             OpenlineageDataset,
+            OperatorLineage,
             OutputStatisticsOutputDatasetFacet,
             SchemaDatasetFacet,
             SchemaField,
             SqlJobFacet,
         )
-        from astro.lineage.extractor import OpenLineageFacets
 
         input_dataset: list[OpenlineageDataset] = []
         output_dataset: list[OpenlineageDataset] = []
-        if (
-            self.output_table.openlineage_emit_temp_table_event() and self.output_table.conn_id
-        ):  # pragma: no cover
+
+        first_table = find_first_table(
+            op_args=self.op_args,  # type: ignore
+            op_kwargs=self.op_kwargs,
+            python_callable=self.python_callable,
+            parameters=self.parameters,  # type: ignore
+            context={},
+        )
+
+        if first_table.openlineage_emit_temp_table_event() and first_table.conn_id:  # pragma: no cover
             input_uri = (
-                f"{self.output_table.openlineage_dataset_namespace()}"
```

Comment:
just curious earlier it was  output_table now why first_table?

Reply:
it was not correct before as in transform the input table is the first table.